### DISCUSSION TITLE
Add Focus Mode (not Zen Mode)

### DISCUSSION THREAD

**Discussion by MarkuBu** [helix-editor/helix#10439] [Ideas]
The Atom Editor has an addon called Focus Mode (not to be confused with Zen Mode from VSCode).

In Focus Mode the current scope (paragraph, loop, function and so on) is normally visible, the other code is dimmed. 

https://github.com/davidleghorn/atom-focus-mode

This should be easily done with tree-sitter. The most interesting function is the scope focus mode. But the Atom addon is limited. I think it could be nice to change the level of highlighting like to show only the current if block or while loop, or the current block plus one level above, or the whole function.

I made some mockup examples with VSCode to show what I mean with the level of highlighting (this is part of the Helix code)

only the for() loop
![Screenshot_20220319_090342](https://user-images.githubusercontent.com/16163058/159113231-8f8db82f-d266-40c0-b75a-8acb63b48056.png)

the match
![Screenshot_20220319_090507](https://user-images.githubusercontent.com/16163058/159113233-329d8a47-a2f8-455a-b47c-8750882175ad.png)

the whole function
![Screenshot_20220319_090533](https://user-images.githubusercontent.com/16163058/159113235-717c44a5-3a18-486f-8761-6ef6e555b2da.png)

**Comment by slinlee** [comment]
For reference, this is what I used in vim: https://github.com/junegunn/limelight.vim
